### PR TITLE
Add `get_all_xxx` methods to core nodes.

### PR DESCRIPTION
**Pull request by jetuk**
Add methods to core nodes to get values for all scenarios. This is more efficient than call the `get_xxx` methods many times.

### CODE REVIEW

File: pywr/_core.pyx [hunk 1]
```diff
@@ -477,6 +477,14 @@ cdef class AbstractNode:
     cpdef double get_cost(self, ScenarioIndex scenario_index) except? -1:
         return 0.0
 
+    cpdef double[:] get_all_cost(self, double[:] out=None):
```
Comment (Batch21) [inline pywr/_core.pyx:480]: Is there a specific reason for having the `out` arg? and does there need to be any check on its shape?

File: pywr/_core.pyx [hunk 2]
```diff
@@ -800,19 +858,54 @@ cdef class StorageInput(BaseInput):
         BaseInput.commit(self, scenario_index, volume)
         self._parent.commit(scenario_index, -volume)
 
+    cpdef commit_all(self, double[:] value):
```
Comment (Batch21) [inline pywr/_core.pyx:861]: I'm guessing these method are just to allow net flow for a storage node to be easily recorded?

**Review by Batch21 — COMMENTED** [review]
Changes look good. Are the pipeline failures all related to xlrd?

File: pywr/_core.pyx (hunk 2 above)
Comment (jetuk) [reply pywr/_core.pyx:861]: It allows the solver to set net flows for all scenarios simultaneously. A `commit_all` method already exists on `AbstractNode`, but these storage nodes need different behaviour to update the parent storage node.

File: pywr/_core.pyx (hunk 1 above)
Comment (jetuk) [reply pywr/_core.pyx:480]: It allows for a pre-allocated array to be given in which the values are set (i.e. avoids the need to reallocate space on every call). A solver would allocate space once per run to accept all data, and then pass those arrays as the `out` argument. This avoids an additional allocation and possibly a copy of the data.

**Comment by jetuk** [comment]
> Changes look good. Are the pipeline failures all related to xlrd?

I think so.

**Comment by jetuk** [comment]
This is now rebased on master so the pipeline should work.

**Review by Batch21 — APPROVED** [review]

**Pull request merged**